### DISCUSSION TITLE
Cannot iterate:  Android BuildConfig.java VERSION_CODE = 2

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#242] [Q&A] [answered]
I've had success deploying the FIRST internal testing product to Play Store.  Now I'm stuck trying to iterate...

After making changes to the App and running the Android > skip export  I get a BuildConfig.java with the VERSION_CODE = 2;

Trying to upload the signed "app-release.aab"  I get Error **“Version code 2 has already been used. Try another version code.”**
I trace this "2" back to the  VERSION_CODE = 2;  But don't know where it is coming from!  My iOS project file has:
Version 0.15.8
Build  7

Can anyone tell me how the Java is getting the 2 - so that I can change it and iterate my App?

**Comment by marcprux** [comment]
I've seen that happen before – I think Fastlane sometimes bumps this automatically when repeatedly trying to upload a version to the Play Store. What does the "Latest Version" report on the Google Play Console page for the app?

I've always just gotten around it by incrementing the number to something higher. Ultimately, it doesn't matter what the number is, as long as it is higher than the most recently submitted version.

**Reply by davidakoontz** [reply]
Version 2... didn't find "Latest Version report.

Found this page:
![Screenshot 2024-09-16 at 1 17 12 PM](https://github.com/user-attachments/assets/e8b3ed2d-47ef-438c-84d4-cd7c4aa9df45)

**Reply by davidakoontz** [reply]
that is right in sync with BuildConfig.java
```
package painlesspassword.module;

public final class BuildConfig {
  public static final boolean DEBUG = Boolean.parseBoolean("true");
  public static final String APPLICATION_ID = "com.lifeworksiq.painless_password";
  public static final String BUILD_TYPE = "debug";
  public static final int VERSION_CODE = 2;
  public static final String VERSION_NAME = "0.15.1";
}

```

**Reply by davidakoontz** [reply]
the 0.15.1 is coming from iOS Version in project file.
I've not got any 2s  ... that I can find.

**Reply by davidakoontz** [reply]
I've changed the iOS Version from 0.15.1  to 0.15.8
<img width="896" alt="Screenshot 2024-09-16 at 1 25 22 PM" src="https://github.com/user-attachments/assets/4158ad13-66d4-4c16-85d8-d848f0a84e88">

Deleted the hidden .build folder and rerun the Xcode build...

The BuildConfig.java continues to hold:
public static final String VERSION_NAME = **"0.15.1"**;

So I conclude that some automated generate code is NOT getting ReGenerated properly...

Since that line is right next to:
public static final int **VERSION_CODE = 2**;

I'm guessing I need to kick some tires to REGENERATE Android code....

Don't know the magic...  SKIP CLEAN   SKIP Regenerate

**Reply by davidakoontz** [reply]
From what I can tell...

> skip export

Does a Build of iOS code.  Makes an *.ipa archive.

Then does an Assemble process on the *.ipa

Then does an Assemble Android  <<<=== I assume this does the **code Generation** ???

But the 0.15.1 has NOT changed in Version_Name!!

❯ pwd
/Users/david/Developer/Fastlane_projects/ppwd/.build/Android/app/generated/source/buildConfig/release/painlesspassword/module
❯ more BuildConfig.java
/**
 * Automatically generated file. DO NOT MODIFY
 */
package painlesspassword.module;

public final class BuildConfig {
  public static final boolean DEBUG = false;
  public static final String APPLICATION_ID = "com.lifeworksiq.painless_password";
  public static final String BUILD_TYPE = "release";
  public static final int VERSION_CODE = 2;
  public static final String VERSION_NAME = "0.15.1";
}


So I conclude there is a CACHE somewhere keeping a OLD version of the iOS Project file.

**Reply by davidakoontz** [reply]
OH - I'm NOT using FASTLANE now.  Could not get it to configure and work... doing hand deploys (and I hate myself for it)!

**Comment by davidakoontz** [comment, marked as answer]
Oh... I think I found the CACHE... it's listed in Skip.env file!

So I updated the Skip.env numbers... and starting down the build / export / sign... highway.  (No Fastlane).

// The configuration file for your Skip App (https://skip.tools).
// Properties specified here are shared between
// Darwin/painlesspassword.xcconfig and Android/settings.gradle.kts
// and will be included in the app's metadata files
// Info.plist and AndroidManifest.xml

// PRODUCT_NAME is the default title of the app, which must match the app's Swift module name
PRODUCT_NAME = painlesspassword

// PRODUCT_BUNDLE_IDENTIFIER is the unique id for both the iOS and Android app
PRODUCT_BUNDLE_IDENTIFIER = com.lifeworksiq.painless-password

// The semantic version of the app
MARKETING_VERSION = 0.15.2

// The build number specifying the internal app version
CURRENT_PROJECT_VERSION = 3

// The package name for the Android entry point, referenced by the AndroidManifest.xml
ANDROID_PACKAGE_NAME = painlesspassword.module
